### PR TITLE
Update whats new and upgrade guide

### PR DESCRIPTION
**Pull request by shaojingli**
@Pessimistress @gnavvy @1chandu

### CODE REVIEW

File: docs/whats-new.md [hunk 1]
```diff
@@ -1,19 +1,32 @@
 # deck.gl v4.1
 
-## CompositeLayer
+Release date: July 28th, 2017
 
-* `CompositeLayer.renderLayers` can now return a nested arrays with `null` values. deck.gl will automatically flatten and filter the array. This is a small convenience that makes the `renderLayers methods in complex composite layers a little more readable.
+## WebGL2 support provided by luma.gl v4
 
+While deck.gl v4.1 is not a backward-compatible minor release, its rendering functionality is backed by luma.gl v4, a WebGL2-enabled rendering framework. One all browsers that supports WebGL2, deck.gl will obtain WebGL2 context and utilize WebGL2 functionalities. To know more about WebGL2, please check [here](https://www.khronos.org/registry/webgl/specs/latest/2.0/).
```
Comment (1chandu) [inline docs/whats-new.md]: nit : One all -> Once all

File: docs/upgrade-guide.md [hunk 2]
```diff
@@ -2,12 +2,45 @@
...
+	...
+	});
+```
+
+new:
+```
+const model = new Model(gl, {
+	vs: vertexShader,
+	fs: fragmentShader,
+	modules: ['fp64', ...],
```
Comment (gnavvy) [inline docs/upgrade-guide.md]: do we have a list of available modules listed somewhere?

File: docs/upgrade-guide.md [hunk 3]
```diff
@@ -2,12 +2,45 @@
 
 ## Upgrading from deck.gl v4 to v4.1
 
-A highly compatible release, the biggest change is that deck.gl v4.1 brings in luma.gl v4. It is a highly compatible release but there are some changes. If you are writing deck.gl layers using luma.gl classes, please refer to the Upgrade Guide for luma.gl.
+deck.gl v4.1 is a backward-compatible release. Most of the functionality and APIs remain unchanged but there are smaller changes that might requires developers' attention if they /**develop their own layers**.
```
Comment (gnavvy) [inline docs/upgrade-guide.md]: backward compatible release with v4.0 (maybe redundant for those who's familiar with semver)

File: docs/whats-new.md (hunk 1 above)
Comment (gnavvy) [inline docs/whats-new.md]: While deck.gl v4.1 is _--not--_ a backward-compatible minor release

**Pull request merged**